classes/rule.php

```diff
@@ -137,6 +137,21 @@ public function load_from_form($form) {
         $this->data_save();
     }
 
+    /**
+     * Load a rule from a trigger object.
+     * @param $trigger
+     */
+    public static function load_rule_from_trigger($trigger) {
+
+        $rule = new rule;
+        $trigger = get_object_vars($trigger);
+
+        foreach ($trigger as $key => $item) {
```

Comment:
This looks good, but can we also add a check using property_exists like we have in load_from_form?